tests/tensor_search/integ_tests/test_no_model.py

```diff
@@ -0,0 +1,163 @@
+import os
+from unittest.mock import patch
+
+from marqo.api.exceptions import InvalidArgError
+from marqo.core.exceptions import IndexNotFoundError
+from marqo.core.models.marqo_index import *
+from marqo.core.models.marqo_index_request import FieldRequest
+from marqo.tensor_search import tensor_search
+from marqo.tensor_search.models.search import SearchContext
+from tests.marqo_test import MarqoTestCase
+from tests.utils.transition import *
+
+
+class TestNoModel(MarqoTestCase):
+
+    structured_index_with_no_model = "structured_index_with_no_model"
+    unstructured_index_with_no_model = "unstructured_index_with_no_model"
+    DIMENSION = 16
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+
+        structured_index_with_no_model_request = cls.structured_marqo_index_request(
+            name=cls.structured_index_with_no_model,
+            model=Model(name="no_model", properties={"dimensions": cls.DIMENSION, "type": "no_model"}, custom=True),
+            fields=[
+                FieldRequest(name='text_field_1', type=FieldType.Text),
+                FieldRequest(name='image_field_1', type=FieldType.ImagePointer),
+                FieldRequest(name="custom_field_1", type=FieldType.CustomVector)
+            ],
+            tensor_fields=["text_field_1", "image_field_1", "custom_field_1"]
+        )
+        unstructured_index_with_no_model_request = cls.unstructured_marqo_index_request(
+            name=cls.unstructured_index_with_no_model,
+            model=Model(name="no_model", properties={"dimensions": cls.DIMENSION, "type": "no_model"}, custom=True),
+        )
+
+        # List of indexes to loop through per test. Test itself should extract index name.
+        cls.indexes = cls.create_indexes([
+            structured_index_with_no_model_request,
+            unstructured_index_with_no_model_request,
+        ])
+
+    def setUp(self) -> None:
+        # Any tests that call add_documents, search, bulk_search need this env var
+        self.device_patcher = patch.dict(os.environ, {"MARQO_BEST_AVAILABLE_DEVICE": "cpu"})
+        self.device_patcher.start()
+
+    def tearDown(self) -> None:
+        self.device_patcher.stop()
+
+    def test_create_index_with_invalid_model_properties(self):
+        """Test to ensure an invalid model properties will block index creation."""
+        try:
+            self.index_management.delete_index_by_name("test_create_index_with_invalid_model_name_or_properties")
+        except IndexNotFoundError:
+            pass
+
+        # Test that creating an index with an invalid model name or properties raises an error
+        test_cases = [
+            ("no_model", {"dimensions": 0, "type": "no_model"}, "invalid dimensions value, can't be 0"),
```

Comment:
Should we include a test case with no `type` provided, or one where the name is `no_model`, but it does not match the type?

Reply:
good catch, added!